src/bug-filing/github/create-github-bug-filing-url.ts

```diff
@@ -2,57 +2,57 @@
 import { CreateIssueDetailsTextData } from './../../common/types/create-issue-details-text-data';
-import { IssueFilingUrlStringUtils } from './../common/issue-filing-url-string-utils';
+import { IssueFilingUrlStringUtils, StringUtils } from './../common/issue-filing-url-string-utils';
 import { GitHubBugFilingSettings } from './github-bug-filing-service';
 
-export function createGitHubBugFilingUrl(
-    settingsData: GitHubBugFilingSettings,
-    bugData: CreateIssueDetailsTextData,
-    environmentInfo: EnvironmentInfo,
-): string {
-    function buildTitle(data: CreateIssueDetailsTextData): string {
-        const standardTags = IssueFilingUrlStringUtils.standardizeTags(data);
-        let prefix = standardTags.join(',');
-        if (prefix.length > 0) {
-            prefix = prefix + ': ';
-        }
-
-        const selectorLastPart = IssueFilingUrlStringUtils.getSelectorLastPart(data.ruleResult.selector);
-
-        return `${prefix}${data.ruleResult.help} (${selectorLastPart})`;
+function buildTitle(stringUtils: StringUtils, data: CreateIssueDetailsTextData): string {
+    const standardTags = stringUtils.standardizeTags(data);
+    let prefix = standardTags.join(',');
+    if (prefix.length > 0) {
+        prefix = prefix + ': ';
     }
 
-    function buildGithubText(data: CreateIssueDetailsTextData): string {
-        const result = data.ruleResult;
-
-        const text = [
-            `**Issue**: \`${result.help}\` ([\`${result.ruleId}\`](${result.helpUrl}))`,
-            ``,
-            `**Target application**: [${data.pageTitle}](${data.pageUrl})`,
-            ``,
-            `**Element path**: ${data.ruleResult.selector}`,
-            ``,
-            `**Snippet**:`,
-            ``,
-            `    ${IssueFilingUrlStringUtils.collapseConsecutiveSpaces(result.snippet)}`,
-            ``,
-            `**How to fix**:`,
-            ``,
-            `${IssueFilingUrlStringUtils.markdownEscapeBlock(result.failureSummary)}`,
-            ``,
-            `**Environment**:`,
-            `${environmentInfo.browserSpec}`,
-            ``,
-            `====`,
-            ``,
-            IssueFilingUrlStringUtils.footer(environmentInfo),
-        ].join('\n');
-
-        return text;
-    }
+    const selectorLastPart = stringUtils.getSelectorLastPart(data.ruleResult.selector);
 
-    const title = buildTitle(bugData);
-    const body = buildGithubText(bugData);
-    const encodedIssue = `/new?title=${encodeURIComponent(title)}&body=${encodeURIComponent(body)}`;
-    return `${settingsData.repository}${encodedIssue}`;
+    return `${prefix}${data.ruleResult.help} (${selectorLastPart})`;
 }
+
+function buildGithubText(stringUtils: StringUtils, environmentInfo: EnvironmentInfo, data: CreateIssueDetailsTextData): string {
+    const result = data.ruleResult;
+
+    const text = [
+        `**Issue**: \`${result.help}\` ([\`${result.ruleId}\`](${result.helpUrl}))`,
```

Comment:
if this is going to be a key value pair, can you create a method that generates the line given a kv pair?

Reply:
I'm not touching this logic as part of this PR... I'm just moving the code out of and static function in a class. Does it make sense to refactor this as part of this PR?